client.go

```diff
@@ -477,13 +487,26 @@ func (c *Client) stream(method, path string, streamOptions streamOptions) error
 	if streamOptions.stderr == nil {
 		streamOptions.stderr = ioutil.Discard
 	}
-	cancelRequest := cancelable(c.HTTPClient, req)
+
+	// make a sub-context so that our active cancellation does not affect parent
+	ctx := streamOptions.context
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	subCtx, cancelRequest := context.WithCancel(ctx)
+	defer cancelRequest()
+
 	if protocol == "unix" {
 		dial, err := c.Dialer.Dial(protocol, address)
 		if err != nil {
 			return err
 		}
-		cancelRequest = func() { dial.Close() }
+		go func() {
+			select {
+			case <-subCtx.Done():
+				dial.Close()
```

Comment:
Hmm there doesn't seem to be any noticeable effect from double Close(). Then I realized there's no need for `defer dial.Close()` either, since `defer cancelRequest()` would always trigger the goroutine to do a single `Close()`.